langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingUtils.java

```diff
@@ -0,0 +1,82 @@
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.Values;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Neo4jEmbeddingUtils {
+    
+    /* not-configurable strings, used in `UNWIND $rows ...` statement */
+    public static final String EMBEDDINGS_ROW_KEY = "embeddingRow";
+    public static final String ID_ROW_KEY = "id";
+    
+    /* default configs */
+    public static final String DEFAULT_EMBEDDING_PROP = "embeddingProp";
+    public static final String PROPS = "props";
+    public static final String DEFAULT_IDX_NAME = "langchain-embedding-index";
+    public static final String DEFAULT_LABEL = "Document";
+    public static final String DEFAULT_METADATA_PREFIX = "metadata.";
+    public static final String DEFAULT_TEXT_PROP = "text";
+    
+
+    public static EmbeddingMatch<TextSegment> toEmbeddingMatch(Neo4jEmbeddingStore store, Record neo4jRecord) {
+        var node = neo4jRecord.get("node").asNode();
+
+        var metaData = new HashMap<String, String>();
+        node.keys().forEach(key -> {
+            if (key.startsWith(store.getMetadataPrefix())) {
+                metaData.put(key.substring(key.indexOf(".") + 1), node.get(key).asString());
+            }
+        });
+
+        Metadata metadata = new Metadata(metaData);
+
+        Value text = node.get(store.getText());
+        TextSegment textSegment = text.isNull()
+                ? null
+                : TextSegment.from(text.asString(), metadata);
+        List<Number> embeddingList = node.get(store.getEmbeddingProperty())
+                .asList(Value::asNumber);
+
+        Embedding embedding = new Embedding(toFloatArray(embeddingList));
+
+        return new EmbeddingMatch<>(neo4jRecord.get("score").asDouble(), node.get(ID_ROW_KEY).asString(), embedding, textSegment);
+    }
+    
+    public static float[] toFloatArray(List<Number> numberList) {
+        float[] embeddingFloat = new float[numberList.size()];
+        int i = 0;
+        for(Number num: numberList) {
+            embeddingFloat[i++] = num.floatValue();
+        }
+        return embeddingFloat;
+    }
+
+    public static Map<String, Object> toRecord(Neo4jEmbeddingStore store, int idx, List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        String id = ids.get(idx);
+        Embedding embedding = embeddings.get(idx);
+
+        Map<String, Object> row = new HashMap<>();
+        row.put(ID_ROW_KEY, id);
+
+        Map<String, Object> properties = new HashMap<>();
+        if (embedded != null) {
+            TextSegment segment = embedded.get(idx);
+            properties.put(store.getText(), segment.text());
+            Map<String, String> metadata = segment.metadata().asMap();
+            metadata.forEach((k, v) -> properties.put(store.getMetadataPrefix() + k, Values.value(v)));
```

Comment:
can we have the default option to be the metadata-keys without a prefix, and only if a prefix is configured it is used?

Reply:
Now the default prefix is `""`